### PR TITLE
Modelgen refactors

### PR DESCRIPTION
**Pull request by erikh**
@shaleman 

I moved everything to template files with a generator to sew them all into a `.go` file for compilation. I think you'll like the result. I also fixed a small issue with the output directory not being created when specified.

More coming, but not tonight. PTAL.

### CODE REVIEW

File: tools/modelgen/generators/templates.go [hunk 1]
```diff
@@ -0,0 +1,355 @@
+package generators
```
Comment (mapuri) [inline tools/modelgen/generators/templates.go]: looks like this file is auto-generated and need not be checked in SCM?

Comment (erikh) [inline tools/modelgen/generators/templates.go]: you're absolutely correct, it does not. I'll remove it before we merge.

**Comment by mapuri** [comment]
LGTM

**Comment by shaleman** [comment]
Nice! Thanks.
LGTM

-sukhesh

**Pull request merged**